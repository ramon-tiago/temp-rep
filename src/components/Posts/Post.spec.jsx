import { render, screen } from '@testing-library/react';
import { Post } from '.';

const props = {
  posts: [
    {
      id: 1,
      title: 'title 1',
      body: 'Hello 1',
      cover: 'img/img1.png',
    },
    {
      id: 2,
      title: 'title 2',
      body: 'Hello 2',
      cover: 'img/img2.png',
    },
    {
      id: 3,
      title: 'title 3',
      body: 'Hello 3',
      cover: 'img/img3.png',
    },
  ],
};

describe('<Posts />', () => {
  it('should render posts', () => {
    render(<Post posts={props.posts} />);
    expect(screen.getAllByRole('heading', { name: /title/i })).toHaveLength(3);

    expect(screen.getAllByRole('img', { name: /title/i })).toHaveLength(3);

    expect(screen.getByRole('img', { name: /title 3/i })).toHaveAttribute('src', 'img/img3.png');

    // expect(screen.getAllByAltText(/body/i))
    //   .toHaveLength(3);
  });

  it('should match snapshot', () => {
    const { container } = render(<Post posts={props.posts} />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
