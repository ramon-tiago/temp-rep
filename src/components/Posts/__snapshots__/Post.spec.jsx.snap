// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Posts /> should match snapshot 1`] = `
<div
  class="posts"
>
  <div
    class="post"
  >
    <img
      alt="title 1"
      class="cover"
      src="img/img1.png"
    />
    <div
      class="post-content"
    >
      <h1>
        title 1
      </h1>
      <p>
        Hello 1
      </p>
    </div>
  </div>
  <div
    class="post"
  >
    <img
      alt="title 2"
      class="cover"
      src="img/img2.png"
    />
    <div
      class="post-content"
    >
      <h1>
        title 2
      </h1>
      <p>
        Hello 2
      </p>
    </div>
  </div>
  <div
    class="post"
  >
    <img
      alt="title 3"
      class="cover"
      src="img/img3.png"
    />
    <div
      class="post-content"
    >
      <h1>
        title 3
      </h1>
      <p>
        Hello 3
      </p>
    </div>
  </div>
</div>
`;
